File: examples/demo-apps/apple_ios/LLaMA/LLaMA/Application/Message.swift
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 */

import Foundation
import SwiftUI

enum MessageType {
  case prompted
  case generated
  case info
}

struct Message: Identifiable, Equatable {
  let id = UUID()
  let dateCreated = Date()
  var dateUpdated = Date()
  var type: MessageType = .prompted
  var text = ""
  var tokenCount = 0
  var image: UIImage?
}

File: hello/message.cpp
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 */

#include "hello/message.h"
#include <flatbuffers/flatbuffers.h>
#include "schema/program_generated.h"

namespace executorch {
  std::string message() {
    auto builder = flatbuffers::FlatBufferBuilder{1024};
    auto msg = builder.CreateString("🏎️ move fast!");
    auto string_offset = executorch_flatbuffer::CreateString(builder, msg);
    builder.Finish(string_offset);
    auto buffer = builder.GetBufferPointer();
    auto string_obj = flatbuffers::GetRoot<executorch_flatbuffer::String>(buffer);
    return string_obj->string_val()->str();
  }
}
